j2k/src/org/jetbrains/kotlin/j2k/CodeConverter.kt

```diff
@@ -115,7 +115,21 @@ class CodeConverter(
         if (needConversion(actualType, expectedType)) {
             val expectedTypeStr = expectedType.canonicalText
             if (expression is PsiLiteralExpression) {
-                if (expectedTypeStr == "float" || expectedTypeStr == "double") {
+                if (actualType.canonicalText == "char" && expression.parent !is PsiExpressionList) {
+                    when (expectedTypeStr) {
```

Comment:
I think that's here should be a better way. 
Better to move such code may be to type converter or so. 
Because it looks like something similar required here #1200

Reply:
Fixed